src/vueWrapper.ts

```diff
@@ -79,7 +79,12 @@ export class VueWrapper<T extends ComponentPublicInstance> {
   }
 
   html() {
-    return this.parentElement.innerHTML
+    // cover cases like <Suspense>, multiple root nodes.
+    if (this.parentElement['__vue_app__']) {
```

Comment:
`__vue_app__` is only present on the root component no? Does it still work if the suspense or multi-root node component is not directly under the app?

Reply:
I would suggest if we could use `this.hasMultipleRoot` ?

Reply:
I have no idea, we should add a test and find out!

Reply:
Seems fine https://github.com/vuejs/vue-test-utils-next/pull/174/commits/fc045f7e83fe2f2a499c26ae6e6a28f2e1d6e3ff#diff-7906b5e8a0a656a314ef4e609e16f76dR54

Is this what you meant @cexbrayat ?

Reply:
I was thinking about an extra layer of component, something like `mount(App)` with `App` just displaying the `Component` you have in your test. Because as is in your test, `Component` is the root comp, so it has the `__vue_app_` attribute I guess? But maybe I'm misunderstanding something: if you're confident that's OK for me!

Reply:
I'm not that confident, better have a test. Can you give me an example of what you mean (doesn't need to be working). The way I read is it something like this:

```js
const Comp = {
  template: `
    <ComponentA>
      <ComponentB>1</ComponentB>
      <ComponentB>2</ComponentB>
      <ComponentB>3</ComponentB>
  </ComponentA>
`
}

const App = {
  template: `
    <ComponentA />
  `
}

mount(App)
```

?

Reply:
Exactly!

Reply:
So this does NOT work

```
  it.only('works', () => {
    const ComponentA = {
      name: 'ComponentA',
      template: `<div class="comp-a"><slot /></div>`
    }
    const ComponentB = {
      name: 'ComponentB',
      template: `<div class="comp-b"><slot /></div>`
    }
    const Main = {
      components: { ComponentA, ComponentB },
      template: `
        <ComponentA>
          <ComponentB>1</ComponentB>
          <ComponentB>2</ComponentB>
          <ComponentB>3</ComponentB>
        </ComponentA>
      `
    }

    const App = {
      components: { Main },
      template: '<Main />'
    }

    const wrapper = mount(App)

    expect(wrapper.getComponent(ComponentB).html()).toBe('<div class="comp-b">1</div>')
```

It cannot find `ComponentB` at all. I don't know why (nothing to do this with PR though).

Reply:
Made issue... https://github.com/vuejs/vue-test-utils-next/issues/180

I still don't fully get how Vue stores the component hierarchy internally. Hm.